### PR TITLE
#162170680 Article share button

### PR DESCRIPTION
**Pull request by frostyblok**
#### What does this PR do?
Allows users to share articles on social media

#### Description of Task to be completed?
* Create component for sharing an article that takes in the article slug and article as props
* Test the component using jest and React-testing-library 

#### How should this be manually tested?
- Clone this repository `git@github.com:andela/balder-ah-frontend.git`
- cd into the directory on your local machine
- On your command line, run `npm install` to install the project npm packages
- Run `npm run start:dev` to start the app
- On your web browser, visit an article page on `/article/slug` and check out the implementation by clicking on the star icons to rate the article on a scale of 1 to 5.
- Run `npm test` to test the app
- Run `npm test -- --coverage` to view the test coverage

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories? (If applicable)
[#162170680](https://www.pivotaltracker.com/story/show/162170680)

#### Screenshots (If applicable)

screen shot 2019-01-25 at 12 53 16 am

screen shot 2019-01-25 at 12 53 44 am

#### Questions:

### CODE REVIEW

File: src/assets/sass/base.scss [hunk 1]
```diff
@@ -114,3 +114,49 @@ h6 {
...
+  padding-bottom: 10px;
+  display: none;
+}
+
+.icon-list-style-show  {
+  animation-duration: 0.7s;
+  animation-name: slidein;
+  display: flex;
+  padding-bottom: 10px;
+  // visibility: visible;
```
Comment (akhilome) [inline src/assets/sass/base.scss]: You forgot to take this out. 🙂

File: src/components/presentation/shareButton/Share.jsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  handleClick = () => {
+    this.setState(prevState => ({
+      toggleShare: !prevState.toggleShare,
+    }));
+  }
+
+  render() {
+    const { REACT_APP_URL } = process.env;
+    const { articleSlug, articleTitle } = this.props;
+    const url = `${REACT_APP_URL}/articles/${articleSlug}`;
```
Comment (akhilome) [inline src/components/presentation/shareButton/Share.jsx:26]: You forgot to add the `REACT_APP_URL` to the `.env.sample` file.

**Review by akhilome — COMMENTED** [review]
Good job delivering on this, @frostyblok.

I left a few concerns below, please go over them.

**Review by akhilome — APPROVED** [review]

File: src/components/presentation/shareButton/Share.jsx (hunk 2 above)
Comment (frostyblok) [reply src/components/presentation/shareButton/Share.jsx:26]: Thank you for spotting this out. I have made the changes!

**Comment by omokaro-faith** [comment]
You'll want to fix conflicts.

File: src/components/presentation/shareButton/Share.jsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  GooglePlusShareButton,
+  GooglePlusIcon,
+} from 'react-share';
+
+class Share extends React.Component {
+  state = {
+    toggleShare: false,
+  }
+
+  handleClick = () => {
```
Comment (luqmanoop) [inline src/components/presentation/shareButton/Share.jsx]: @frostyblok Awesome work. Quick one... event handler names should be descriptive enough for someone reading your code without having to look at where it's been used.

`handleClick` is kind of ambiguous don't you think? Handle click for what? A text, link, form?

I think this should be refactored to `onShareClick`, `onShareIconClick` or `handleShareIconClick`

**Review by luqmanoop — CHANGES_REQUESTED** [review]
Hi @frostyblok, nice work. I requested a minimal change [here](https://github.com/andela/balder-ah-frontend/pull/24/files#r251449503). Please take a look at it and resolve ASAP.

File: src/components/presentation/shareButton/Share.test.jsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import { render, cleanup } from 'react-testing-library';
+import Share from './Share';
+
+const props = {
+  articleSlug: 'ogbuke-kosilowo-345542',
+  articleTitle: 'Kiging to the core',
+};
+
+describe('Share component test', () => {
+  afterEach(cleanup);
```
Comment (luqmanoop) [inline src/components/presentation/shareButton/Share.test.jsx]: Good job with the cleanup. However, we're already handling for this in our `setupTest.js` file

Comment (luqmanoop) [reply src/components/presentation/shareButton/Share.test.jsx]: Please remove it.

File: src/components/presentation/shareButton/Share.jsx (hunk 3 above)
Comment (frostyblok) [reply src/components/presentation/shareButton/Share.jsx]: Good spot Chief! I will work on that ASAP!

File: src/components/presentation/shareButton/Share.jsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  }
+
+  render() {
+    const { REACT_APP_URL } = process.env;
+    const { articleSlug, articleTitle } = this.props;
+    const url = `${REACT_APP_URL}/articles/${articleSlug}`;
+    const title = articleTitle;
+    const { toggleShare } = this.state;
+    return (
+      <React.Fragment>
```
Comment (oluwajuwon) [inline src/components/presentation/shareButton/Share.jsx:30]: Hi @frostyblok, I can see you're not handling for sharing via emails as well, any particular reason for that?

**Review by oluwajuwon — COMMENTED** [review]
Good job @frostyblok 👍 
I left a comment on your PR, please check it out

File: src/components/presentation/shareButton/Share.jsx (hunk 5 above)
Comment (frostyblok) [reply src/components/presentation/shareButton/Share.jsx:30]: No standout reason for that, I just included that to allow the users to share articles on more platforms.

**Review by luqmanoop — APPROVED** [review]

**Review by okoroemeka — APPROVED** [review]

File: src/components/presentation/shareButton/Share.test.jsx (hunk 4 above)
Comment (frostyblok) [reply src/components/presentation/shareButton/Share.test.jsx]: Done!

**Comment by omokaro-faith** [comment]
@frostyblok please resolve conflicts.

File: src/components/presentation/shareButton/Share.jsx (hunk 5 above)
Comment (oluwajuwon) [reply src/components/presentation/shareButton/Share.jsx:30]: Alright then 👍

**Review by oluwajuwon — APPROVED** [review]

**Pull request merged**